File: frontend/types/audit.ts
export type DocumentRecord = {
  id: string;
  filename: string;
  entity: string | null;
  doc_date: string | null;
  status: "processing" | "indexed" | "failed" | "deleted" | "duplicate";
  page_count: number;
  chunk_count: number;
  created_at: string;
};

export type DocumentPage = {
  documents: DocumentRecord[];
  total: number;
  page: number;
  page_size: number;
  has_next: boolean;
};

export type AuditEvent = {
  id: string;
  session_id: string;
  query: string;
  answer: string | null;
  status: "answered" | "pending_review" | "rejected";
  confidence_score: number | null;
  model_name: string | null;
  prompt_version: string | null;
  pii_detected: boolean;
  created_at: string;
};

export type AuditLogPage = {
  entries: AuditEvent[];
  total: number;
  page: number;
  page_size: number;
  has_next: boolean;
};

export type ReviewItem = {
  id: string;
  session_id: string;
  query: string;
  draft_answer: string;
  confidence_score: number | null;
  citations: Array<{ document_id?: string; page_number?: number; similarity?: number }>;
  status: "pending" | "approved" | "rejected" | "overridden";
  created_at: string;
};

export type ReviewItemPage = {
  items: ReviewItem[];
  total: number;
  page: number;
  page_size: number;
  has_next: boolean;
};

File: backend/app/schemas/audit.py
from datetime import datetime

from pydantic import BaseModel


class AuditLogEntry(BaseModel):
    id: str
    session_id: str
    user_id: str | None = None
    query: str
    answer: str | None = None
    citations: list[dict]
    confidence_score: float | None = None
    prompt_version: str | None = None
    model_name: str | None = None
    trace_id: str | None = None
    status: str
    review_item_id: str | None = None
    pii_detected: bool
    validation_passed: bool
    validation_errors: list[str]
    created_at: datetime


class AuditLogFilter(BaseModel):
    status: str | None = None
    date_from: datetime | None = None
    date_to: datetime | None = None
    min_confidence: float | None = None
    max_confidence: float | None = None
    pii_only: bool = False


class AuditLogPage(BaseModel):
    entries: list[AuditLogEntry]
    total: int
    page: int
    page_size: int
    has_next: bool
